mlpf/pytorch_delphes/LRP/hooks.py

```diff
@@ -0,0 +1,106 @@
+from glob import glob
+import sys, os
+import os.path as osp
+import pickle, math, time, numba, tqdm
+import numpy as np
+import pandas as pd
+import sklearn
+from sklearn.metrics import accuracy_score, confusion_matrix
+import matplotlib, mplhep
+matplotlib.use("Agg")
+import matplotlib.pyplot as plt
+
+#Check if the GPU configuration has been provided
+import torch
+use_gpu = torch.cuda.device_count()>0
+multi_gpu = torch.cuda.device_count()>1
+
+try:
+    if not ("CUDA_VISIBLE_DEVICES" in os.environ):
+        import setGPU
+        if multi_gpu:
+            print('Will use multi_gpu..')
+            print("Let's use", torch.cuda.device_count(), "GPUs!")
+        else:
+            print('Will use single_gpu..')
+except Exception as e:
+    print("Could not import setGPU, running CPU-only")
+
+#define the global base device
+if use_gpu:
+    device = torch.device('cuda:0')
+else:
+    device = torch.device('cpu')
+
+import torch_geometric
+import torch.nn as nn
+import torch.nn.functional as F
+import torch_geometric.transforms as T
+from torch.nn import Sequential as Seq, Linear as Lin, ReLU
+from torch_scatter import scatter_mean
+from torch_geometric.nn.inits import reset
+from torch_geometric.data import Data, DataLoader, DataListLoader, Batch
+from torch_geometric.nn import GravNetConv
+from torch.utils.data import random_split
+import torch_cluster
+
+sys.path.insert(1, '../')
+sys.path.insert(1, '../../../plotting/')
+sys.path.insert(1, '../../../mlpf/plotting/')
```

Comment:
I think this pattern could give some issues with python modules. You should have proper modules defined by placiong `__init__.py` files in the appropriate locations, and be able to do relative imports as in e.g. https://github.com/jpata/particleflow/blob/master/mlpf/tfmodel/model.py#L8